storage/client.go

```diff
@@ -424,9 +424,9 @@ func (c Client) execInternalJSON(verb, url string, headers map[string]string, bo
 	return respToRet, nil
 }
 
-func readResponseBody(resp *http.Response) ([]byte, error) {
-	defer resp.Body.Close()
-	out, err := ioutil.ReadAll(resp.Body)
+func readBody(body io.ReadCloser) ([]byte, error) {
```

Comment:
I would rename to `readAndCloseBody` to make it more clear. Maybe add a link to the godoc in which it is explained why this must be like that.
Other that that LGTM.